packages/common/src/types.ts

```diff
@@ -106,3 +106,50 @@ export type IBBox = number[][];
 export type IBatch = any[];
 
 export type IBatchTransform = (value: any) => any;
+
+export interface IFormItem extends IItem {
+  type: "Form"
+};
+
+export interface IFormModel extends IModel {
+  item: IFormItem;
+  data?: {
+    [propName: string]: any;
+  }
+};
+
+export interface IFeatureServiceItem extends IItem {
```

Comment:
Should some of these types and methods be pushed into arcgis-rest-js or solutions.js?

Reply:
`Hub.js` is a dependency of `Solutions.js`, so probably can't move these types there. I could see an argument for moving `IFormItem` and `IFeatureServiceItem` to `arcgis-rest-js`, although these survey sharing methods would be the only thing using them at the moment. I'd need to take a closer look at any `arcgis-rest-portal` and `hub.js` peer dependencies, suspect we may need to account for breaking API changes by updating to latest.

Reply:
Additionally, this level of granularity wrt typing is probably not necessary, I _could_ also just refactor to just use the existing `IModel` and `IItem` interfaces vs `IFormItem`, `IFormModel`, `IFeatureServiceItem`, & `IFeatureServiceModel`.